### PR TITLE
Add redux store using ducks pattern

### PR DESCRIPTION
**Pull request by sabetech**
In this pull request
- Created a directory structure following the ducks pattern
- Created the actions and reducers for the application
- Exported the reducers to be made available to be invoked

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
```
Comment (omar-labana) [inline src/redux/configureStore.js]: - [ ] Good job creating the initial storage config. The data store should be created and initialized using the redux function `createStore`. Kindly take another look at the milestone requirements.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @sabetech, 

Excellent work so far! 👌
You have implemented almost all the required features without any bug or errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (sabetech) [reply src/redux/configureStore.js]: Oh I missed that. Thanks

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+import books from './books/books';
+import categories from './categories/categories';
+
```
Comment (SajjadAhmad14) [inline src/redux/configureStore.js:4]: Please use `createStore` to create a Redux store as it's the requirement and was requested by the previous code reviewer.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Albert,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/configureStore.js (hunk 2 above)
Comment (sabetech) [reply src/redux/configureStore.js:4]: Thanks for the clarification.

**Review by brytebee — APPROVED** [review]
Hi, @sabetech👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 




Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**